### PR TITLE
feat: implement the `productCollections` method

### PR DESCRIPTION
**Pull request by NWRichmond**
## Why are these changes introduced?

Addresses [ENG-8325](https://nacelle.atlassian.net/browse/ENG-8325) 

> Implement the `productCollections` method

## What is this pull request doing?

1. Adds the `productCollections` method to `@nacelle/storefront-sdk-plugin-commerce-queries` to match `@nacelle/storefront-sdk@1.x`'s `productCollections` method.
2. Refactors some `console` side effects in other methods with new utilities.
3. Refactors some tests to simplify and improve clarity.

## How to Test

1. Check out the `ENG-8325-implement-the-productcollections-method` branch.
2. Navigate to `packages/storefront-sdk-plugins/commerce-queries`.
4. `npm run coverage` - all tests should pass with 100% coverage.

ENG-8325-implement-the-productcollections-method branch all test passing with 100% coverage

5. `npm run build` - there should be no build errors.

From there, you can test in a local project via `npm link`. In my local testing, I found that the `productCollections` method works as expected. For example, the following request returned the expected number of collections & products in the collection:

```ts
const { data } = await client.productCollections({
  maxReturnedEntries: 1,
  maxReturnedEntriesPerCollection: 2,
});
```

@nacelle:storefront-sdk-plugin-commerce-queries demo productCollections

## Checklist

- [x] This Pull Request aligns with `nacelle-js`' [Code of Conduct](https://github.com/getnacelle/nacelle-js/blob/main/CODE_OF_CONDUCT.md#code-of-conduct).
- [x] You can follow your own "How to Test" instructions and get the expected result.
- [x] Screenshots, videos, etc. in the PR description are free of brand names and sensitive details.
- [x] Created a [changeset](https://github.com/changesets/changesets) (if the change should appear in a changelog).

[ENG-8325]: https://nacelle.atlassian.net/browse/ENG-8325?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by KrisQ — APPROVED** [review]
Works!

Screen Shot 2023-02-01 at 10 09 07 AM
Screen Shot 2023-02-01 at 10 12 07 AM

File: packages/storefront-sdk-plugins/commerce-queries/src/utils/messages.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import type { dataFetchingMethods } from './constants.js';
```
[the remark this answers is not in the document]
Comment (mdarrik) [reply packages/storefront-sdk-plugins/commerce-queries/src/utils/messages.ts:1]: It's definitely a little weird. But one way to look at it is that we're only using it in a type context. (eg we use it to define a type, and never access it in a way that JS would use the value.

File: packages/storefront-sdk-plugins/commerce-queries/src/utils/constants.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+export const dataFetchingMethods = [
+	'content',
+	'navigation',
+	'products',
+	'productCollections',
+	'productCollectionEntries',
+	'query',
+	'spaceProperties'
+] as const;
```
Comment (mdarrik) [inline packages/storefront-sdk-plugins/commerce-queries/src/utils/constants.ts:9]: Is there a reason we're using this as an array of values vs a string literal type? 


Eg this over something like this: 


```ts
export type dataFetchingMethods = 'content' | 'navigation' | 'products' | 'productCollections' | 'productCollectionEntries' | 'query' | 'spaceProperties'
```

Comment (NWRichmond) [reply packages/storefront-sdk-plugins/commerce-queries/src/utils/constants.ts]: Refactored in https://github.com/getnacelle/nacelle-js/pull/323/commits/02b418f144290bf999f9abd3355f2703094870dd.

File: packages/storefront-sdk-plugins/commerce-queries/src/index.ts [hunk 3]
```diff
@@ -203,6 +219,73 @@ function commerceQueriesPlugin<TBase extends WithStorefrontQuery & WithConfig>(
...
+				return responseData;
+			}
+
+			return {
+				data: await (this as unknown as StorefrontClient)['applyAfter'](
+					'productCollections',
+					// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+					responseData.data!
+				)
+			} as StorefrontResponse<ContentEdge[] | Content[]>;
```
Comment (mdarrik) [inline packages/storefront-sdk-plugins/commerce-queries/src/index.ts:287]: Is this the right type for productCollections?

Comment (NWRichmond) [reply packages/storefront-sdk-plugins/commerce-queries/src/index.ts]: Not in the slightest! Nice catch. Fixed in https://github.com/getnacelle/nacelle-js/pull/323/commits/f86d3693ba12f7b8b59cbe1ac4979aa475ffa20b. Also added some new tests to check that all methods are returning the expected type.

**Review by mdarrik — APPROVED** [review]
LGTM! Thanks!!

**Pull request merged**